src/main/java/org/spongepowered/api/block/tile/Tile.java

```diff
@@ -48,4 +48,24 @@
      */
     BlockLoc getBlock();
 
+    /**
+     * Checks for whether the tile is currently valid or not.
```

Comment:
There's some hackery that goes on behind the scenes for switching blocks under tile entities and the like (for certain mods), because validation does not depend on the state of the underlying blocks. There was some debate in IRC over whether we should actually expose TE validation, and we decided that it was a better idea to expose it than to keep it hidden.